### DISCUSSION TITLE
Error while setting up the remote computer

### DISCUSSION THREAD

**Discussion by Ezio-droid** [aiidateam/aiida-core#5347] [Q&A] [answered]
Hello Everyone,
I am a newbie to aiida and its ecosystem. So I am just starting to explore aiida. For my quantum espresso calculation, I started with one of the sections of aiida documentation "How to run external codes", because I want to submit the job to supercomputers from my local workstation using aiida. So I followed every step as per the documentation. But after configuration when I am testing it, I am getting the following error:

Info: Testing computer<narval> for user<Tanay.Sahu@inrs.ca>...
* Opening connection... 02/09/2022 08:33:53 PM <43461> aiida.transport.SshTransport: [ERROR] Error connecting to 'narval.computecanada.ca' through SSH: [SshTransport] encountered RSA key, expected OPENSSH key, connect_args were: {'username': 'tanay11', 'port': 22, 'look_for_keys': True, 'key_filename': '/home/tanay/.ssh/aiida', 'timeout': 60, 'allow_agent': True, 'proxy_jump': '', 'proxy_command': '', 'compress': True, 'gss_auth': False, 'gss_kex': False, 'gss_deleg_creds': False, 'gss_host': 'narval.computecanada.ca'}
[FAILED]: Error while trying to connect to the computer
  Full traceback:
  Traceback (most recent call last):
    File "/home/tanay/miniconda3/envs/aiida/lib/python3.9/site-packages/aiida/cmdline/commands/cmd_computer.py", line 473, in computer_test
      with transport:
    File "/home/tanay/miniconda3/envs/aiida/lib/python3.9/site-packages/aiida/transports/transport.py", line 125, in __enter__
      self.open()
    File "/home/tanay/miniconda3/envs/aiida/lib/python3.9/site-packages/aiida/transports/plugins/ssh.py", line 520, in open
      self._client.connect(self._machine, **connection_arguments)
    File "/home/tanay/miniconda3/envs/aiida/lib/python3.9/site-packages/paramiko/client.py", line 435, in connect
      self._auth(
    File "/home/tanay/miniconda3/envs/aiida/lib/python3.9/site-packages/paramiko/client.py", line 766, in _auth
      raise saved_exception
    File "/home/tanay/miniconda3/envs/aiida/lib/python3.9/site-packages/paramiko/client.py", line 678, in _auth
      key = self._key_from_filepath(
    File "/home/tanay/miniconda3/envs/aiida/lib/python3.9/site-packages/paramiko/client.py", line 588, in _key_from_filepath
      key = klass.from_private_key_file(key_path, password)
    File "/home/tanay/miniconda3/envs/aiida/lib/python3.9/site-packages/paramiko/pkey.py", line 249, in from_private_key_file
      key = cls(filename=filename, password=password)
    File "/home/tanay/miniconda3/envs/aiida/lib/python3.9/site-packages/paramiko/ed25519key.py", line 58, in __init__
      pkformat, data = self._read_private_key("OPENSSH", f)
    File "/home/tanay/miniconda3/envs/aiida/lib/python3.9/site-packages/paramiko/pkey.py", line 354, in _read_private_key
      raise SSHException(
  paramiko.ssh_exception.SSHException: encountered RSA key, expected OPENSSH key
Warning: 1 out of 0 tests failed

I really can't understand the source of this error. Is this due to the information I am entering while "setting up the computer" is wrong or something else? Hoping for some positive response.

Thanks in advance.

**Comment by ltalirz** [comment, marked as answer]
Looking at https://github.com/paramiko/paramiko/issues/1612 this seems to be a misleading paramiko exception that simply means that your key could not be loaded (e.g. because it has a passphrase, because it doesn't have the right file permissions, because maybe you passed the public key instead of the private one, ...)

**Reply by Ezio-droid** [reply]
Thanks, Talirz for the quick response. I went to that post before, but I really cannot understand what has to be done. I would really appreciate it if you can tell me more specifically what can be done.

**Reply by ltalirz** [reply]
AiiDA uses the paramiko library to connect via SSH. You can try using it directly, e.g. like in https://stackoverflow.com/a/32485653/1069467

(note that one needs to pass the _private_ key rather than the public one, as the first comment on the answer points out).

If you get that to work, you should be able to get it to work in AiiDA as well.

**Reply by mbercx** [reply]
Perhaps the first things to check is if the (I'm assuming _private_) key in `/home/tanay/.ssh/aiida` has the right permissions (`600` (`-rw-------`)), and if you set a passphrase when generating the key. If so, you can follow the documentation on how to use an SSH agent to store the passphrase:

https://aiida.readthedocs.io/projects/aiida-core/en/latest/howto/ssh.html#using-passphrase-protected-keys-via-an-ssh-agent

I highly recommend the approach with the `load-singlesshagent.sh` described there, this way you shouldn't have to worry about re-entering passphrases after a restart for example.

**Reply by Ezio-droid** [reply]
Thanks @Italirz and @mbercx I figured out the way for that error using yours suggestions. Now I am facing the new problem and I am not sure whether it is right to post it in this thread or not.
So I was setting up quantum espresso code using "verdi code setup" and faced the following error:
**Critical: Unable to store the Code: The local executable '/home/tanay/Desktop/Softwares/DFT/q-e-qe-6.5/bin/pw.x' is not in the list of files of this code**
As per my understanding, is it due to the fact that aiida does not support QE-6.5 version ? Or it is due to some other reason?

Thanks and Regards

**Comment by ltalirz** [comment]
Hi Eizo, here is the code where this message comes from
https://github.com/aiidateam/aiida-core/blob/2f5170d8a8b62c1091024ccaa0e5faa34f2be51f/aiida/orm/nodes/data/code.py#L287

Unfortunately there is a bit of a naming issue in AiiDA with the "local" and the "remote" codes.
Even if pw.x is installed on your machine you'll want to use a "remote" code.

"Local code" means "store the executable in the database and copy it to the target machine when executing. It was originally meant for simple postprocessing scripts but as far as we are aware it is not really used, and I have advocated for removing this functionality.
